debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSchema.java

```diff
@@ -298,45 +301,52 @@ public boolean applyDdl(SourceInfo source, String databaseName, String ddlStatem
+                    // being modified by the DDL statements (since the DDL statements can have fully-qualified names).
+                    // Therefore, we have to look at each statement to figure out which database it applies and then
+                    // record the DDL statements (still in the same order) to those databases.
+
+                    if (!ddlChanges.isEmpty() && ddlChanges.applyToMoreDatabasesThan(databaseName)) {
+
+                        // We understood at least some of the DDL statements and can figure out to which database they apply.
+                        // They also apply to more databases than 'databaseName', so we need to apply the DDL statements in
+                        // the same order they were read for each _affected_ database, grouped together if multiple apply
+                        // to the same _affected_ database...
+                        ddlChanges.groupStatementStringsByDatabase((dbName, ddl) -> {
+                            if (filters.databaseFilter().test(dbName) || dbName == null || "".equals(dbName)) {
+                                if (dbName == null) dbName = "";
+                                statementConsumer.consume(dbName, ddlStatements);
+                            }
+                        });
+                    } else if (filters.databaseFilter().test(databaseName) || databaseName == null || "".equals(databaseName)) {
+                        if (databaseName == null) databaseName = "";
+                        statementConsumer.consume(databaseName, ddlStatements);
+                    }
                 }
-            }
 
-            // Record the DDL statement so that we can later recover them if needed. We do this _after_ writing the
-            // schema change records so that failure recovery (which is based on of the history) won't lose
-            // schema change records.
-            try {
-                dbHistory.record(source.partition(), source.offset(), databaseName, tables, ddlStatements);
-            } catch (Throwable e) {
-                throw new ConnectException(
-                        "Error recording the DDL statement(s) in the database history " + dbHistory + ": " + ddlStatements, e);
+                // Record the DDL statement so that we can later recover them if needed. We do this _after_ writing the
+                // schema change records so that failure recovery (which is based on of the history) won't lose
+                // schema change records.
+                try {
+                    if (!storeOnlyMonitoredTablesDdl || changes.stream().anyMatch(filters().tableFilter()::test)) {
+                        dbHistory.record(source.partition(), source.offset(), databaseName, tables, ddlStatements);
+                    } else {
+                        logger.debug("Changes for DDL '{}' were filtered and not recorded in schema history", ddlStatements);
```

Comment:
s/schema history/database history (topic)/

schema history topic = the public one
DB history topic = the internal one (which is written here)

Reply:
Done,